packages/lesswrong/server/resolvers/commentResolvers.ts

```diff
@@ -192,4 +200,82 @@ augmentFieldsDict(Comments, {
       return (post && post.contents && post.contents.version) || "1.0.0";
     },
   },
+  doppelComments: {
+    resolveAs: {
+      type: '[DoppelComment]',
+      resolver: async (dbComment, context: ResolverContext) => {
+        const user = await Users.findOne({ _id: dbComment.userId });
+        if (!user) return []
+        if (user.karma < 1e4) return []
+        const doppelComments = await runFragmentQuery({
+          fragmentName: 'DoppelCommentsFragment',
+          terms: { view: 'doppelCommentsForComment', commentId: dbComment._id },
+          collectionName: 'DoppelComments',
+        })
```

Comment:
erm... can't this just be a query on the table by commentId?

Reply:
This resolver is going to run for every comment, including on post-pages with hundreds of comments. I don't know whether this is fast enough for that use case, but I strongly suspect it isn't.